src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,556 @@
+        # TODO: This should probably be a part of the hash for the "id" of the task!
+        self.train_config = train_config
+
+        if os.path.exists(checkpoint_file):
+            self.net, self.h = load_task_network(checkpoint_file)
+        else:
+            logger.info(
+                f"Checkpoint file {checkpoint_file} doesn't exist: re-training the meta-model."
+            )
+            # TODO: Figure out where/how to set the logger config.
+            logger.info(f"Task hash params: {task_hash_params}")
+            print(f"Task hash params:", json.dumps({k: str(v) for k, v in task_hash_params.items()}, indent="\t"))
+
+            checkpoint_file.parent.mkdir(exist_ok=True, parents=True)
+            # Need to re-train the meta-model and sample this task.
+            self.net, self.h = get_task_network(
+                input_dir, benchmark, rng, task_id, config=train_config
+            )
+            save_task_network(checkpoint_file, benchmark, self.net, self.h)
+
+    @property
+    def hash(self) -> str:
+        # TODO: Return a unique "hash"/id/key for this task
+
+        if is_dataclass(self):
+            return compute_identity(**asdict(self))
+        return compute_identity(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            **self.fixed_values,
+            seed=self.seed,
+        )
+
+    @property
+    def space(self) -> Dict[str, str]:
+        space = spaces[self.benchmark].copy()
+        for k in self.fixed_values:
+            if k in space:
+                space.pop(k)
+        return space
+
+    @property
+    def full_space(self) -> Dict[str, str]:
+        return spaces[self.benchmark]
+
+    def __call__(self, x: np.ndarray = None, **kwargs):
+        if x is not None:
+            assert not kwargs, "can't pass x and kwargs at the same time!"
+            if isinstance(x, list):
+                return [self(x_i, **kwargs) for x_i in x]
+            hp_dict = self.to_hp_dict(x)
+        else:
+            hp_dict = kwargs
+
+        logger.debug(
+            f"received x={x}, kwargs: {kwargs}, fixed values: {self.fixed_values}"
+        )
+        hp_dict.update(self.fixed_values)
+
+        # TODO: We don't use `self.space` here, because that attribute is
+        # truncated whenever a value if fixed with `kwargs`.
+        # assert set(kwargs.keys()) == set(self.full_space.keys()), f"{set(kwargs.keys())} != {set(spaces[self.benchmark].keys())}"
+        # NOTE: We assume here that keys are in order of insertion.
+        x = np.array([hp_dict[key] for key in self.full_space.keys()])
+        p = np.concatenate((x, self.h))[None, :]
+
+        with torch.no_grad():
+            # BUG in pybnn/util/layers.py, where they have self.log_var be a
+            # DoubleTensor while x is a FloatTensor! Here we overwrite the
+            # forward method, instead of editing the code directly.
+            from pybnn.util.layers import AppendLayer
```

Comment:
Maybe this should be done outside __call__ as it only need to be done once. It should also be contained in a separate function.